text/0131-pass-native-sdk-spans-to-hybrid.md

```diff
@@ -0,0 +1,181 @@
+- Start Date: 2024-02-08
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/131
+- RFC Driver: [Philipp Hofmann](https://github.com/philipphofmann)
+- RFC Status: draft
+
+# Summary
+
+This RFC aims to find a solution for passing auto-instrumented native SDK spans up to the hybrid SDKs.
+
+# Motivation
+
+The native SDKs instrument file IO operations, DB queries, HTTP requests, and more, but this
+information doesn't end up in the hybrid transactions or anywhere else. Our hybrid SDKs instrument
+most of the mentioned operations on the hybrid layer, but they lack information on how long the
+actual native operations take. It could happen that one DB query from the hybrid layer gets
+converted into multiple native DB queries. Such information would be insightful for our users.
+Furthermore, there could be native-only operations that our hybrid SDKs currently don't capture. Therefore,
+we want to pass native spans up to the native layers so users know what's happening on the native layer.
```

Comment:
in a world of ingesting only spans, do we need to pass them across layers? 
Would it dumping straight to the transport be enough while having them on the same trace?

Reply:
That's an excellent question. We want to establish a parent-child relationship between some spans, which works best when passing spans from native to RN because, on the backend, it's tricky if the spans don't arrive there in the same envelope. I will update the RFC to explain this.

Reply:
I added your suggestion as option 4. Feel free to directly commit pros and cons, @bruno-garcia.